core/src/main/scala/org/apache/spark/deploy/PythonRunner.scala

```diff
@@ -37,8 +38,15 @@ object PythonRunner {
     val pythonFile = args(0)
     val pyFiles = args(1)
     val otherArgs = args.slice(2, args.length)
+    val sparkConf = new SparkConf()
     val pythonExec =
-      sys.env.getOrElse("PYSPARK_DRIVER_PYTHON", sys.env.getOrElse("PYSPARK_PYTHON", "python"))
+      if (sparkConf.contains(SparkLauncher.PYSPARK_DRIVER_PYTHON)) {
```

Comment:
The following might be slightly easier to follow (and is closer to the code in the launcher):

```
Seq(sparkConf.getOption(SparkLauncher.PYSPARK_DRIVER_PYTHON),
  sparkConf.get(SparkLauncher.PYSPARK_PYTHON),
  sys.env.get("PYSPARK_DRIVER_PYTHON"),
  sys.env.get("PYSPARK_PYTHON""),
  Some("python")).flatten.head
```